src/REL/d/t/d_t_col_bomb_shield.cpp

```diff
@@ -1,9 +1,50 @@
 #include "d/t/d_t_col_bomb_shield.h"
+#include "common.h"
+#include "d/a/obj/d_a_obj_base.h"
+#include "d/a/d_a_player.h"
+#include "d/col/cc/d_cc_s.h"
+#include "m/m_mtx.h"
+#include "rvl/MTX/mtx.h"
 
 SPECIAL_ACTOR_PROFILE(COL_BOMSLD, dTgColBombShield, fProfile::COL_BOMSLD, 0x12, 0, 0);
 
 STATE_DEFINE(dTgColBombShield, Wait);
 
+bool dTgColBombShield::createHeap() {
+    return true;
+}
+
+int dTgColBombShield::create() {
+    0.0f; // cool
+    CREATE_ALLOCATOR(dTgColBombShield);
+    static const f32 offset = 0.1f;
+    mScale.x += offset;
+    mScale.y += offset;
+    mScale.z += offset;
+    mColMtx.transS(position.x, position.y + mScale.y * 0.5f - 0.05f, position.z);
+    mColMtx.ZXYrotM(rotation.x, rotation.y, rotation.z);
+    mMtx_c tmp;
+    PSMTXScale(tmp, mScale.x, mScale.y, mScale.z);
+    mColMtx += tmp;
+    mStateMgr.changeState(StateID_Wait);
+    return SUCCEEDED;
+}
+
+int dTgColBombShield::doDelete() {
+    return SUCCEEDED;
+}
+
+int dTgColBombShield::actorExecute() {
+    if (dAcPy_c::LINK != nullptr && roomid == dAcPy_c::LINK->roomid) {
+        dCcS::GetInstance()->SetArea(&mColMtx);
```

Comment:
Documentation opportunity: This is literally the only caller of this function in the entire game

Reply:
Yep I recall seeing this, it still didn’t give me enough info as I cannot think with matrices enough yet xD. I wish we had a good way of visualizing this stuff